DeploymentAgentReconfigurationScripts/TFS2018 Update2/HardwareMoveRemoveOldAgent.ps1

```diff
@@ -0,0 +1,178 @@
+    throw "Run command in an administrator PowerShell prompt"
+};
+
+If ($PSVersionTable.PSVersion -lt (New-Object System.Version("3.0"))){
+    throw "The minimum version of Windows PowerShell that is required by the script (3.0) does not match the currently running version of Windows PowerShell."
+}
+
+
+# Auto detect existing agent folder
+if ($existingAgentFolder -eq ""){
+    $ServiceNamePrefix = 'vstsagent*'
+    $agentServices = Get-Service -Name $ServiceNamePrefix
+    if ($agentServices.Count -eq 1){
+        $serviceName = $agentServices[0].Name;
+        $service = get-wmiobject -query "select * from win32_service where name = `'$serviceName`'";
+        $serviceExePath = $service.pathname;
+        # $serviceExePath will have value like "C:\vstsAgent2\A8\bin\AgentService.exe"
+        $existingAgentFolder = $serviceExePath.Substring(1, $serviceExePath.Length - 22);
+    }
+}
+
+if ($existingAgentFolder -eq ""){
+    throw "Not able to auto detect existing agent folder. Provide the existingAgentFolder as input parameter. You can find it in agent capabilities UI as Agent.HomeDirectory.  Generally it is in C:\vstsagent folder.";
+}
+
+cd $existingAgentFolder;
+
+if (-not (Test-Path '.\.agent')){
+    throw "No agent installed in this path. Please run this script from Agent Home Directory. Generally it is in C:\vstsagent folder."
+}
+
+
+# Collect information about the existing agent
+$sourceTFSUrl = "NoAbleToReadAgentFile"
+$collectionName = "NoAbleToReadAgentFile"
+$projectName = "NoAbleToReadAgentFile"
+$deploymentGroupId = "NoAbleToReadAgentFile"
+$deploymentGroupName = "NoAbleToReadAgentFile"
+$agentName = "NoAbleToReadAgentFile"
+$tags = "NoAbleToReadAgentFile"
+$deploymentMachineId = "NoAbleToReadAgentFile"
+
+foreach($line in Get-Content .\.agent) {
+    $token = $line.split('"');
+
+    if ($token[1] -eq 'serverUrl'){
+        $sourceTFSUrl = $token[3];
+        if ($sourceTFSUrl.StartsWith($targetTFSUrl)){
+            Write-Verbose -Verbose "Agent is already configured to the TFS $targetTFSUrl";
+            return 0;
+        }
+    }
+
+    if ($token[1] -eq 'collectionName'){
+        $collectionName = $token[3];
+    }
+
+    if ($token[1] -eq 'projectId'){
+        $projectName = $token[3];
+    }
+
+    if ($token[1] -eq 'deploymentGroupId'){
+        $deploymentGroupId = $token[2].Substring(2, $token[2].Length - 3);
+    }
+
+    if ($token[1] -eq 'agentName'){
+        $agentName = $token[3];
+    }
+}
+
+# Get the Deployment group name
+$encodedPat = [System.Convert]::ToBase64String([System.Text.Encoding]::UTF8.GetBytes(":$patToken"))
+$getDeploymentGroupUrl = $targetTFSUrl + '/' + $collectionName + '/' + $projectName + '/_apis/distributedtask/deploymentgroups/' + $deploymentGroupId + '?api-version=5.0-preview.1'
+ 
+$dg = Invoke-RestMethod -Uri $getDeploymentGroupUrl -Method Get -Headers @{Authorization = "Basic $encodedPat"} -ContentType "application/json"
```

Comment:
if we can write this in functions that will be better, fine with this as well

Reply:
Keeping this as readable as possible for user